components/custom_status/custom_status_emoji.test.tsx

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import {mount, shallow} from 'enzyme';
+import React from 'react';
+
+import configureStore from 'redux-mock-store';
+import {Provider} from 'react-redux';
+
+import * as CustomStatusSelectors from 'selectors/views/custom_status';
+
+import CustomStatusEmoji from './custom_status_emoji';
+
+jest.mock('selectors/views/custom_status');
+
+describe('components/custom_status/custom_status_emoji', () => {
+    const mockStore = configureStore();
+    const store = mockStore({});
+
+    it('should match snapshot', () => {
+        const wrapper = shallow(
+            <Provider store={store}>
+                <CustomStatusEmoji/>
+            </Provider>,
+        );
+
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it('should match snapshot with props', () => {
+        const wrapper = shallow(
+            <Provider store={store}>
+                <CustomStatusEmoji
+                    emojiSize={34}
+                    showTooltip={true}
+                    tooltipDirection='bottom'
+                />
+            </Provider>,
+        );
+
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it('should not render when EnableCustomStatus in config is false', () => {
+        (CustomStatusSelectors.isCustomStatusEnabled as jest.Mock).mockReturnValue(false);
```

Comment:
@hmhealey I think this test case is fine right now. Here's why:

1. We should mock the selectors in components so that we don't have to update all components that use a particular selector when the store hierarchy of a reducer changes.
2. We can have separate test cases that cover whether each selector detects changes in the correct reducer/store hierarchy.

Thoughts?

Reply:
Yeah, that's very true. I guess I was thinking of the easy case where a selector is just returning something simple, but even in those cases, we still end up needing to make up a large part of the state tree, and that's not something the people implementing a component need to care about.

We haven't really ran into this before since class components are tested without the redux connection, but that also means the connector often doesn't get tested, so this covers that. The more I think about it, this does seem like the right way to test the component, and the selector can be tested on its own.